src/server/document-controller.js

```diff
@@ -65,13 +55,19 @@ class DocumentController {
     });
 
     router.get('/docs/:docKey', needsPermission(permissions.VIEW_DOCS), async (req, res) => {
-      const revisions = await this.documentService.getAllDocumentRevisionsByKey(req.params.docKey);
-      if (!revisions.length) {
+      const doc = await this.documentService.getDocumentByKey(req.params.docKey);
+      return res.redirect(301, urls.getDocUrl(doc.key, doc.slug));
+    });
+
+    router.get('/docs/:docKey/*', async (req, res) => {
```

Comment:
Shouldn't we check if `*` matches the actual slug of the doc (and redirect if not), otherwise we have infinite URLs per document (I'm thinking of Google indexing, there probably should be a 301 to the correct "canonical" URL)?

Reply:
Okay, good point, will do :).

Reply:
Fixed